server/chat-plugins/scavengers.ts

```diff
@@ -800,16 +795,16 @@ export class ScavengerHunt extends Rooms.RoomGame {
 	}
 
 	tryRunQueue(roomid: RoomID) {
-		if (this.room.scavgame || this.room.scavSettings.scavQueueDisabled) return; // don't run the queue for child games.
+		if (this.room.scavgame || this.room.scavSettings?.scavQueueDisabled) return; // don't run the queue for child games.
 		// prepare the next queue'd game
 		if (this.room.scavQueue && this.room.scavQueue.length) {
 			setTimeout(() => {
 				const room = Rooms.get(roomid) as ChatRoom;
-				if (!room || room.game || !room.scavQueue.length) return;
+				if (!room || room.game || !room.scavQueue?.length) return;
 
 				const next = room.scavQueue.shift();
 				if (!next) return; // shouldn't happen, but it makes typescript happy.
```

Comment:
```suggestion
				const next = room.scavQueue.shift()!;
```

(this is the correct way to make TS happy)